bundles/org.openhab.binding.anthem/src/main/java/org/openhab/binding/anthem/internal/handler/AnthemConnectionManager.java

```diff
@@ -0,0 +1,433 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.anthem.internal.handler;
+
+import static org.openhab.binding.anthem.internal.AnthemBindingConstants.*;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.InterruptedIOException;
+import java.io.OutputStreamWriter;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.anthem.internal.AnthemConfiguration;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.IncreaseDecreaseType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AnthemConnectionManager} is responsible for handling the connection
+ * to the Anthem AV processor.
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+@NonNullByDefault
+public class AnthemConnectionManager {
+    private final Logger logger = LoggerFactory.getLogger(AnthemConnectionManager.class);
+
+    private static final long POLLING_INTERVAL_SECONDS = 900L;
+    private static final long POLLING_DELAY_SECONDS = 10L;
+
+    private ScheduledExecutorService scheduler;
+
+    private final AnthemHandler handler;
+    private AnthemConfiguration configuration;
+
+    private @Nullable Socket socket;
+    private @Nullable BufferedWriter writer;
+    private @Nullable BufferedReader reader;
+
+    private AnthemCommandParser messageParser;
+
+    private final BlockingQueue<AnthemCommand> sendQueue = new LinkedBlockingQueue<>();
+
+    private @Nullable Future<?> asyncInitializeTask;
+    private @Nullable ScheduledFuture<?> connectRetryJob;
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private @Nullable Thread senderThread;
+    private @Nullable Thread readerThread;
+
+    private int reconnectIntervalMinutes;
+    private int commandDelayMsec;
+
+    private boolean zone1PreviousPowerState;
+    private boolean zone2PreviousPowerState;
+
+    public AnthemConnectionManager(AnthemHandler handler, AnthemConfiguration configuration) {
+        messageParser = new AnthemCommandParser(this);
+        this.configuration = configuration;
+        this.handler = handler;
```

Comment:
This class seems very tightly coupled with the handler. Actually it's almost like a partial handler class doing much of the same things you would normally expect a handler to do. It's documented as a class handling the connection to the device, so maybe consider if you could reduce it to do just that, and decouple it from the handler logic.

Reply:
Yes, it is. I went down a bit of a rabbit hole after I got started with it. I may just move all the functionality back into the handler class, assuming that's also acceptable to you.

Reply:
Of course, only meant to trigger a consideration about this, not a strict requirement. So I would probably either make them loosely coupled (interface/callback) or simply move the logic back into the handler directly as you suggest, as then it would at least be more transparent and life-cycle would be more obvious.

Reply:
It's actually pretty easy to get rid of the `AnthemConnectionManager` completely by moving it all back into the handler. And it will simplify things considerably, so it feels like the right thing to do.

Reply:
AnthemCommunicationManager class refactored into AnthemHandler

Reply:
I will need one more review iteration going through `AnthemCommandParser`.

Reply:
@jlaur Friendly reminder that you intended to do a final pass through this. No rush, just didn't want it to get lost.

Reply:
Sorry, thanks for the ping. I had another look, and I still believe it would be better to decouple `AthemCommandParser` from the handler. If the `parseMessage` method would return an object containing channel and state, there would be no need for the callbacks anymore, and it would be easy to add unit tests. This is just a suggestion. If you prefer not to refactor this, I won't block the PR because of this, and we can merge as is.

Reply:
I see your point, and I'm not against refactoring this. But I won't be able to test the changes for a least another month or so (the Anthem is in another location and I won't be there until the end of April). I also need to figure out how to handle the cases other than channel/state updates (e.g. property updates, triggering another command to send, etc.). I'll likely need to create several different response object types (one for channel/state updates, one for property updates, etc.). So it's a bit more involved than it was to decouple the sender/reader threads.

I also need to make at least one other enhancement (such as a new channel for sending custom commands), which I also won't be able to test for another month or so.

Therefore, I'd propose that you merge this as is. Then when I make the other change(s), I can also do the decoupling. Let me know if you're ok with that approach.

Reply:
Sounds like a good plan, let's merge it now in good time before next milestone, and you can make additional enhancements in new PR's when you have the opportunity.